thavalon-server/src/game/state.rs

```diff
@@ -198,20 +200,43 @@ macro_rules! in_phases {
 
 impl GameStateWrapper {
     /// Creates the [`GameState`] wrapper for a new game.
-    pub fn new(game: Game) -> GameStateWrapper {
+    pub fn new(game: Game) -> ActionResult {
         let first_proposer = &game.proposal_order()[0];
         let phase = Proposing::new(first_proposer.clone());
 
+        let mut effects = vec![Effect::Broadcast(Message::ProposalOrder(
+            game.proposal_order.clone(),
+        ))];
+
+        for player in game.players.iter() {
+            effects.push(Effect::Send(
+                player.name.clone(),
+                Message::RoleInformation {
+                    details: game.info[&player.name].clone(),
+                },
+            ));
+        }
+
+        // Send NextProposal last to move client to the proposal phase after
+        // receiving role information.
+        effects.push(Effect::Broadcast(Message::NextProposal {
+            proposer: first_proposer.clone(),
+            mission: 1,
+            proposals_made: 0,
+            max_proposals: game.spec.max_proposals,
```

Comment:
Would it be useful to send clients the `GameSpec`(or a subset of it) at the start of the game?

Reply:
I'm not sure it's useful now, though if we supported additional variants in the future it might be. I don't think we have anything directly referencing the game rules in the front end.